db_normalizer.py

```diff
@@ -0,0 +1,221 @@
+import sys
+import re
+import sqlite3
+import pandas as pd
+
+def repair_referential_IC(
+    food_licensee_inspections,
+    food_licensees, 
+    food_inspection_violations,
+    food_violations,
+    food_locations,
+    connection):
+
+    template= f"""
+    BEGIN TRANSACTION;
```

Comment:
FYI, I found the issue. It turns out that it is a known sqlite bug. 

If there is a collection, C1, with a M2O field referencing another collection, C2 (i.e., table C1 has a foreign key referencing table C2), then, if you try to add another relational field to collection C2, a "Foreign Key constraint failed" error occurs. In our case, we have Food Inspection Violations table referencing Food Licensee Inspections which also references Food Licensees table. Unfortunately, there is no fix for this bug so we can't use primary key and foreign key  to enforce the constraints. Github issue: [https://github.com/directus/directus/issues/10874](https://github.com/directus/directus/issues/10874)

Sure, I will add the queries.

Reply:
Linked error is *not* a bug in `SQLite`, but with some other tool's usage of it. I know for a fact that it's possible to build exactly the schema I specified, because I did it, myself; it's just that the order of operations matters. I'll upload my version for you to see for yourself.

Reply:
Reference [normalize.sql](https://uillinoisedu-my.sharepoint.com/:u:/g/personal/dmcguire_illinois_edu/EZQvzbvChgdMpSZcU4iEExEBHn7rtwxbPx-apSSW4aNazQ?e=H4IYtD) and [Food_Inspections.sqlite](https://uillinoisedu-my.sharepoint.com/:u:/g/personal/dmcguire_illinois_edu/EYmFYZi99TRBpMuzHUYzrKwBsVSTrlW1RMbIfjiR8MN2QA?e=h4jb2b).